Framework/API/src/InfoComponentVisitor.cpp

```diff
@@ -39,6 +40,8 @@ void InfoComponentVisitor::registerComponentAssembly(
 
   m_ranges.emplace_back(std::make_pair(detectorStart, detectorStop));
 
+  // Record the ID -> index mapping
+  m_componentIdToIndexMap[assembly.getComponentID()] = m_componentIds.size();
```

Comment:
Why are you creating this in the visitor now? It is still also done in `ComponentInfo` itself?